app/src/controllers/Api.tsx

```diff
@@ -0,0 +1,31 @@
+import axios, { CancelToken } from 'axios';
+import { baseURL } from "../globals/variables";
+
+//TODO remove the memory leak from async function by using axios cancel functionality 
```

Comment:
Whats the memory leak issue?

Reply:
Was due to the lack of CancelTokens and was throwing a lovely error